plugins/kubernetes/app/models/kubernetes/release_doc.rb

```diff
@@ -43,12 +43,15 @@ def resource_template
     end
 
     def resources
-      @resources ||= resource_template.map do |t|
-        Kubernetes::Resource.build(
-          t, deploy_group,
-          autoscaled: kubernetes_role.autoscaled,
-          delete_resource: delete_resource
-        )
+      @resources ||= begin
+        resources = resource_template.map do |t|
+          Kubernetes::Resource.build(
+            t, deploy_group,
+            autoscaled: kubernetes_role.autoscaled,
+            delete_resource: delete_resource
+          )
+        end
+        resources.sort_by { |r| Kubernetes::RoleConfigFile::DEPLOY_SORT_ORDER.index(r.kind) }
```

Comment:
```suggestion
        resources.sort_by { |r| Kubernetes::RoleConfigFile::DEPLOY_SORT_ORDER.index(r.kind) || 999 }
```

Reply:
used the size instead of a hardcoded number